### DISCUSSION TITLE
How to change land use during long-term crop rotations?

### DISCUSSION THREAD

**Discussion by KeLiu7** [APSIMInitiative/ApsimX#8755] [Q&A]
Dear all,

Do you know how to change the land use during the crop rotations ? For example, wheat-maize rotations during 1985-2000 then RedClover and PlantainForage rotations from 2000 to 2019? 

my current sims does not work well. I have attached my sims there. Could you please help to solve this problem? @par456 @hol353 Many thanks

Something like below:
![image](https://github.com/APSIMInitiative/ApsimX/assets/72141690/f355840d-64f0-4276-9414-25839a22bf2c)

[Rotation examples.zip](https://github.com/APSIMInitiative/ApsimX/files/14595110/Rotation.examples.zip)

**Comment by par456** [comment]
To make this work I just had to add a start and end date for allowing Maize and Red Clover to sow in their manager scripts:
![image](https://github.com/APSIMInitiative/ApsimX/assets/130029141/64e81146-9608-48b2-896b-73918eeb9c24)
![image](https://github.com/APSIMInitiative/ApsimX/assets/130029141/f821ec64-7a24-4ef1-b9f5-6bc2f2aadd27)
![image](https://github.com/APSIMInitiative/ApsimX/assets/130029141/08e31415-9c65-4341-b750-3ff2a81fd080)

That way when the rotation is in summer fallow, it will also branch to either Maize or Red Clover based on what year it currently is.

To test that it was working, I had to add (and edit) a Crop Soil node under physical for both Ryegrass and Red Clover, so make sure you update those with the correct values for your soil.

![image](https://github.com/APSIMInitiative/ApsimX/assets/130029141/5e1dfe41-70cd-41e4-bf65-87202ce498c2)

I also added a write message statement to the Rotation Manager, manager script that just prints out what state the rotation is in each day, you can see those in the summary log.

Attached is the file I made with these changes:

[Rotation examples.zip](https://github.com/APSIMInitiative/ApsimX/files/14629923/Rotation.examples.zip)

**Reply by KeLiu7** [reply]
Many thanks Andrew. This is super helpful!!!

**Reply by KeLiu7** [reply]
Hi Andrew @par456 ,  I ran into another issue though. RedClover and PlantainForage don't have EndPhase states in their phenology, meaning that the simulation will never transition off their states because they are never ready for harvesting. We might need to come up with some sort of rule to end these crops and add it to their manager scripts because otherwise they will stick around indefinitely.  

I tried to add the start and end time for RedClover and PlantainForageManager respectively.  It does not work once there is pasture grown on the ground, then the other one will not grow. Could you please have another check? Many thanks!

**Comment by KeLiu7** [comment]
> To make this work I just had to add a start and end date for allowing Maize and Red Clover to sow in their manager scripts: ![image](https://private-user-images.githubusercontent.com/130029141/313523889-64e81146-9608-48b2-896b-73918eeb9c24.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTA5OTE5MjcsIm5iZiI6MTcxMDk5MTYyNywicGF0aCI6Ii8xMzAwMjkxNDEvMzEzNTIzODg5LTY0ZTgxMTQ2LTk2MDgtNDhiMi04OTZiLTczOTE4ZWViOWMyNC5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwMzIxJTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDMyMVQwMzI3MDdaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT0xOGQ4YWM0OGI0NGJlYzA5ZjdlZDRiM2I5Y2UxNmU0MWI4NmMzZjllZWUxZjYwMDJjY2QwOGZlMmE1MGNjMTc2JlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.U27KgAZW9jaOwlxAbnrZelzjspvOzkJRNNTFsjcoAik) ![image](https://private-user-images.githubusercontent.com/130029141/313523972-f821ec64-7a24-4ef1-b9f5-6bc2f2aadd27.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTA5OTE5MjcsIm5iZiI6MTcxMDk5MTYyNywicGF0aCI6Ii8xMzAwMjkxNDEvMzEzNTIzOTcyLWY4MjFlYzY0LTdhMjQtNGVmMS1iOWY1LTZiYzJmMmFhZGQyNy5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwMzIxJTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDMyMVQwMzI3MDdaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT01M2EyMDk5OTJmYjYzOTU3OTUyYjQwYzgyY2I0MjY2ZjFhYjkxYzQxNjg1NDBiZjIyYTE1YTJkMDNkNjU1OWU4JlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.o7F7RXXg_EvC_n47x8RSV5EXA0AO8URo_9mJMoqLxT0) ![image](https://private-user-images.githubusercontent.com/130029141/313524025-08e31415-9c65-4341-b750-3ff2a81fd080.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTA5OTE5MjcsIm5iZiI6MTcxMDk5MTYyNywicGF0aCI6Ii8xMzAwMjkxNDEvMzEzNTI0MDI1LTA4ZTMxNDE1LTljNjUtNDM0MS1iNzUwLTNmZjJhODFmZDA4MC5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwMzIxJTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDMyMVQwMzI3MDdaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT1iMDExN2IyZDBlMDE2NTM3OGNhY2JjZjljMWNmOWJhMzEzZDY0ZjBjYzYxYTNhMjJkNTc2ZDAzZDMzNTAzMWMzJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.fxqSRaPGRHDWUTh52Xlvuq-UcQHqXLHiad92ARILoMs)
> 
> That way when the rotation is in summer fallow, it will also branch to either Maize or Red Clover based on what year it currently is.
> 
> To test that it was working, I had to add (and edit) a Crop Soil node under physical for both Ryegrass and Red Clover, so make sure you update those with the correct values for your soil.
> 
> ![image](https://private-user-images.githubusercontent.com/130029141/313524384-5e1dfe41-70cd-41e4-bf65-87202ce498c2.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTA5OTE5MjcsIm5iZiI6MTcxMDk5MTYyNywicGF0aCI6Ii8xMzAwMjkxNDEvMzEzNTI0Mzg0LTVlMWRmZTQxLTcwY2QtNDFlNC1iZjY1LTg3MjAyY2U0OThjMi5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwMzIxJTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDMyMVQwMzI3MDdaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT1lNzc3ZGVmMTUzYTgzZTFhNzBmODg2YThmNGMwN2YwZWE5MWE0Y2M2NTIyNTE4MGVmOTdlYTM3ZGQ5ZjU1NzlmJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.q55yBQgITRNHDL5AqIkoWQYDSz6mrcQv3-xFWl29k3U)
> 
> I also added a write message statement to the Rotation Manager, manager script that just prints out what state the rotation is in each day, you can see those in the summary log.
> 
> Attached is the file I made with these changes:
> 
> [Rotation examples.zip](https://github.com/APSIMInitiative/ApsimX/files/14629923/Rotation.examples.zip)



> To make this work I just had to add a start and end date for allowing Maize and Red Clover to sow in their manager scripts: ![image](https://private-user-images.githubusercontent.com/130029141/313523889-64e81146-9608-48b2-896b-73918eeb9c24.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTA5OTE5MjcsIm5iZiI6MTcxMDk5MTYyNywicGF0aCI6Ii8xMzAwMjkxNDEvMzEzNTIzODg5LTY0ZTgxMTQ2LTk2MDgtNDhiMi04OTZiLTczOTE4ZWViOWMyNC5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwMzIxJTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDMyMVQwMzI3MDdaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT0xOGQ4YWM0OGI0NGJlYzA5ZjdlZDRiM2I5Y2UxNmU0MWI4NmMzZjllZWUxZjYwMDJjY2QwOGZlMmE1MGNjMTc2JlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.U27KgAZW9jaOwlxAbnrZelzjspvOzkJRNNTFsjcoAik) ![image](https://private-user-images.githubusercontent.com/130029141/313523972-f821ec64-7a24-4ef1-b9f5-6bc2f2aadd27.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTA5OTE5MjcsIm5iZiI6MTcxMDk5MTYyNywicGF0aCI6Ii8xMzAwMjkxNDEvMzEzNTIzOTcyLWY4MjFlYzY0LTdhMjQtNGVmMS1iOWY1LTZiYzJmMmFhZGQyNy5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwMzIxJTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDMyMVQwMzI3MDdaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT01M2EyMDk5OTJmYjYzOTU3OTUyYjQwYzgyY2I0MjY2ZjFhYjkxYzQxNjg1NDBiZjIyYTE1YTJkMDNkNjU1OWU4JlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.o7F7RXXg_EvC_n47x8RSV5EXA0AO8URo_9mJMoqLxT0) ![image](https://private-user-images.githubusercontent.com/130029141/313524025-08e31415-9c65-4341-b750-3ff2a81fd080.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTA5OTE5MjcsIm5iZiI6MTcxMDk5MTYyNywicGF0aCI6Ii8xMzAwMjkxNDEvMzEzNTI0MDI1LTA4ZTMxNDE1LTljNjUtNDM0MS1iNzUwLTNmZjJhODFmZDA4MC5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwMzIxJTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDMyMVQwMzI3MDdaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT1iMDExN2IyZDBlMDE2NTM3OGNhY2JjZjljMWNmOWJhMzEzZDY0ZjBjYzYxYTNhMjJkNTc2ZDAzZDMzNTAzMWMzJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.fxqSRaPGRHDWUTh52Xlvuq-UcQHqXLHiad92ARILoMs)
> 
> That way when the rotation is in summer fallow, it will also branch to either Maize or Red Clover based on what year it currently is.
> 
> To test that it was working, I had to add (and edit) a Crop Soil node under physical for both Ryegrass and Red Clover, so make sure you update those with the correct values for your soil.
> 
> ![image](https://private-user-images.githubusercontent.com/130029141/313524384-5e1dfe41-70cd-41e4-bf65-87202ce498c2.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTA5OTE5MjcsIm5iZiI6MTcxMDk5MTYyNywicGF0aCI6Ii8xMzAwMjkxNDEvMzEzNTI0Mzg0LTVlMWRmZTQxLTcwY2QtNDFlNC1iZjY1LTg3MjAyY2U0OThjMi5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwMzIxJTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDMyMVQwMzI3MDdaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT1lNzc3ZGVmMTUzYTgzZTFhNzBmODg2YThmNGMwN2YwZWE5MWE0Y2M2NTIyNTE4MGVmOTdlYTM3ZGQ5ZjU1NzlmJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.q55yBQgITRNHDL5AqIkoWQYDSz6mrcQv3-xFWl29k3U)
> 
> I also added a write message statement to the Rotation Manager, manager script that just prints out what state the rotation is in each day, you can see those in the summary log.
> 
> Attached is the file I made with these changes:
> 
> [Rotation examples.zip](https://github.com/APSIMInitiative/ApsimX/files/14629923/Rotation.examples.zip)
